SchoolProject/src/main/java/Menus/TeacherMenu.java

```diff
@@ -0,0 +1,155 @@
+                    break;
+            }
+        }
+        scanner.close();
+    }
+
+    void menuOption1(){
+        Scanner scanner = new Scanner(System.in);
+        boolean acc=false;
+        students = new StudentRepository();
+        List<Student> student = null;
+        String edu="";
+        while (!acc){
+            System.out.println("Please Select an Education Level");
+            for(EducationLevel Edu:EducationLevel.values()){
+                System.out.println("  - "+Edu);
+            }
+            edu = scanner.nextLine().toUpperCase();
+            if(EducationLevel.HIGH_SCHOOL.exists(edu)){
+                student = students.getStudentByEducationLevel(edu);
+                acc=true;
+            }
+        }
+        System.out.println("=================== List of Students by "+edu+" ===================");
+        int count =1;
+        if(student.size()>0){
+            for(Student st : student){
+                System.out.println(count+". Name: "+st.getName());
+                System.out.println("   Age: "+st.getAge());
+                System.out.println("   Education Level: "+st.getEducationLevel());
+                System.out.println("   Subjects: ");
+                for(Subject sub : st.getSubject()){
+                    System.out.println("     - Name: "+sub.getsubject()+" - Grade: "+sub.getGrade());
+                }
+                System.out.println("-------------------------------------------------------------------");
+                count+=1;
+            }
+        }else{
+            System.out.println("There are no Students on: "+edu);
+        }
+        scanner.close();
+    }
+    
+    void menuOption2(){
+        Scanner scanner = new Scanner(System.in);
+        students = new StudentRepository();
+        boolean acce=false;
+        boolean acc=false;
+        boolean ac=false;
+        String edu="";
+        String grade="";
+        while(!acce){
+            System.out.println("Please enter the student code");
+            String code = scanner.nextLine();
+            AccessValidator accessValidator = new AccessValidator();
+            if(!accessValidator.verifyStudentAccess(code)){
+            }else{
+            Student student =students.getStudentByCode(code);
+            String el=student.getEducationLevel();
+            System.out.println("=============================== STUDENT ===============================");
+            System.out.println("Name: "+student.getName());
+            System.out.println("Age: "+student.getAge());
+            System.out.println("Education Level: "+student.getEducationLevel());
+            System.out.println("   Subjects: ");
+            for(Subject sub : student.getSubject()){
+                System.out.println("     - Name: "+sub.getsubject()+" - Grade: "+sub.getGrade());
+            }
+            System.out.println("-------------------------------------------------------------------");
+            while (!acc){
```

Comment:
Having two while in a block is not correct, it will cause some memory leaks in the code, try to do not to use these kinds of sentences.